meson.build

```diff
@@ -23,142 +23,183 @@ project(
-  if cc.has_member('struct timespec', 'tv_sec', prefix : '#include <time.h>')
-    conf.set('HAVE_STRUCT_TIMESPEC', 1)
+  if config.get('HAVE_TIME_H', 0) == 1
+    if compiler.has_member('struct timespec', 'tv_sec', prefix : '#include <time.h>')
+      config.set('HAVE_STRUCT_TIMESPEC', 1)
+    endif
   endif
-endif
 
-foreach f : ['calloc', 'exit', 'fprintf', 'free', 'longjmp', 'siglongjmp',
-             'malloc', 'memcpy', 'memset', 'printf', 'setjmp', 'signal',
-             'strsignal', 'strcmp', 'clock_gettime']
-  if cc.has_function(f)
-    conf.set('HAVE_@0@'.format(f.underscorify().to_upper()), 1)
-  endif
-endforeach
+  foreach f : ['calloc', 'exit', 'fprintf', 'free', 'longjmp', 'siglongjmp',
+               'malloc', 'memcpy', 'memset', 'printf', 'setjmp', 'signal',
+               'strsignal', 'strcmp', 'clock_gettime']
+    if compiler.has_function(f)
+      config.set('HAVE_@0@'.format(f.underscorify().to_upper()), 1)
+    endif
+  endforeach
 
-if host_machine.system() == 'windows'
+  if machine.system() == 'windows'
   foreach f : ['_vsnprintf_s', '_vsnprtinf', '_snprintf_s', '_snprintf']
-    if cc.has_function(f)
-      conf.set('HAVE_@0@'.format(f.underscorify().to_upper()), 1)
+    if compiler.has_function(f)
+      config.set('HAVE_@0@'.format(f.underscorify().to_upper()), 1)
     endif
   endforeach
   foreach f : ['snprintf', 'vsnprintf']
-    if cc.has_header_symbol('stdio.h', f)
-      conf.set('HAVE_@0@'.format(f.underscorify().to_upper()), 1)
+    if compiler.has_header_symbol('stdio.h', f)
+      config.set('HAVE_@0@'.format(f.underscorify().to_upper()), 1)
     endif
   endforeach
 else
   foreach f : ['snprintf', 'vsnprintf']
-    if cc.has_function(f)
-      conf.set('HAVE_@0@'.format(f.underscorify().to_upper()), 1)
+    if compiler.has_function(f)
+      config.set('HAVE_@0@'.format(f.underscorify().to_upper()), 1)
     endif
   endforeach
 endif
 
-if host_machine.system() == 'windows'
-  if cc.compiles('''
-      __declspec(thread) int tls;
+  if machine.system() == 'windows'
+    if compiler.compiles('''
+        __declspec(thread) int tls;
 
-      int main(void) {
-        return 0;
-      }''',
-      name : 'Thread Local Storage')
-    conf.set('HAVE_MSVC_THREAD_LOCAL_STORAGE', 1)
-  endif
-else
-  if cc.compiles('''
-      __thread int tls;
-
-      int main(void) {
-        return 0;
-      }''',
-      name : 'Thread Local Storage')
-    conf.set('HAVE_GCC_THREAD_LOCAL_STORAGE', 1)
+        int main(void) {
+          return 0;
+        }''',
+        name : 'Thread Local Storage')
+      config.set('HAVE_MSVC_THREAD_LOCAL_STORAGE', 1)
+    endif
+  else
+    if compiler.compiles('''
+        __thread int tls;
+
+        int main(void) {
+          return 0;
+        }''',
+        name : 'Thread Local Storage')
+      config.set('HAVE_GCC_THREAD_LOCAL_STORAGE', 1)
+    endif
   endif
-endif
 
-dep_rt = cc.find_library('rt', required : false)
+  dep_rt = compiler.find_library('rt', required : false)
```

Comment:
Is `dep_rt` actually used anywhere? I couldn't find it referenced.
I know you didn't introduce it, the reason I am asking however is that in the loop this variable is overwritten on the second iteration so this might have broken.

Reply:
Doing a grep, I see no other uses in the repository. I will just remove this.